apps/src/code-studio/progressRedux.js

```diff
@@ -109,7 +143,7 @@ export default function reducer(state = initialState, action) {
     // extract fields we care about from action
     return {
       ...state,
-      currentLevelId: action.currentLevelId,
+      currentLevelId: state.currentLevelId || action.currentLevelId,
```

Comment:
Any chance this might have unintended side effects? how often do we dispatch the INIT_PROGRESS action?

Reply:
It's a healthy concern.  It's dispatched once when the header is first built at page load, using static data baked into the page source, and again each time the user opens the `HeaderPopup` menu to see all levels (at the completion of the asynchronous `/api/user_progress` network request that is triggered by opening that popup).  We want to use `action.currentLevelId` that first time, since it's all we have and is correct at the time, and then `state.currentLevelId` for the subsequent times, just in case the user has switched level without a page reload.

Reply:
Gotcha, thanks! This seems like a potential use case for GraphQL, maybe we would only request certain parts of the progress model when opening the header popup